servers/services/src/main/java/org/projectnessie/services/authz/AccessChecker.java

```diff
@@ -0,0 +1,132 @@
+   * @param ref The {@link NamedRef} to check
+   * @throws AccessControlException When the permission to create a Branch/Tag is not granted.
+   */
+  void canCreateReference(AccessContext context, NamedRef ref) throws AccessControlException;
+
+  /**
+   * Checks whether the given role/principal is allowed to assign the given Branch/Tag to a Hash.
+   *
+   * @param context The context carrying the principal information.
+   * @param ref The {@link NamedRef} to check
+   * @throws AccessControlException When the permission to assign the given Branch/Tag to a Hash is
+   *     not granted.
+   */
+  void canAssignRefToHash(AccessContext context, NamedRef ref) throws AccessControlException;
+
+  /**
+   * Checks whether the given role/principal is allowed to delete a Branch/Tag.
+   *
+   * @param context The context carrying the principal information.
+   * @param ref The {@link NamedRef} to check
+   * @throws AccessControlException When the permission to delete a Branch/Tag is not granted.
+   */
+  void canDeleteReference(AccessContext context, NamedRef ref) throws AccessControlException;
+
+  /**
+   * Checks whether the given role/principal is allowed to read entries content for the given
+   * Branch/Tag.
+   *
+   * @param context The context carrying the principal information.
+   * @param ref The {@link NamedRef} to check
+   * @throws AccessControlException When the permission to read entries content is not granted.
+   */
+  void canReadEntries(AccessContext context, NamedRef ref) throws AccessControlException;
+
+  /**
+   * Checks whether the given role/principal is allowed to list the commit log for the given
+   * Branch/Tag.
+   *
+   * @param context The context carrying the principal information.
+   * @param ref The {@link NamedRef} to check
+   * @throws AccessControlException When the permission to list the commit log is not granted.
+   */
+  void canListCommitLog(AccessContext context, NamedRef ref) throws AccessControlException;
+
+  /**
+   * Checks whether the given role/principal is allowed to commit changes against the given
+   * Branch/Tag.
+   *
+   * @param context The context carrying the principal information.
+   * @param ref The {@link NamedRef} to check
+   * @throws AccessControlException When the permission to commit changes is not granted.
+   */
+  void canCommitChangeAgainstReference(AccessContext context, NamedRef ref)
+      throws AccessControlException;
+
+  /**
+   * Checks whether the given role/principal is allowed to read an entity value as defined by the
+   * {@link ContentsKey} for the given Branch/Tag.
+   *
+   * @param context The context carrying the principal information.
+   * @param ref The {@link NamedRef} to check
+   * @param key The {@link ContentsKey} to check
+   * @throws AccessControlException When the permission to read an entity value is not granted.
+   */
+  void canReadEntityValue(AccessContext context, NamedRef ref, ContentsKey key)
```

Comment:
What is the difference between canReadEntityValue and canReadEntries ? Also is canViewReferences a more controlled access - i.e only view the list of references but not necessarily read the contents ?

Reply:
`canReadEntries` refers to https://github.com/nastra/nessie/blob/40bc1672eab1295c1eaa76ed65d50f145df91d4e/model/src/main/java/org/projectnessie/model/EntriesResponse.java#L27-L49 (which holds the `ContentsKey`) and `canReadEntityValue` refers to the actual `Contents` (https://github.com/nastra/nessie/blob/52a6141fb3d3b538ecaf1c39006e6160832ec166/model/src/main/java/org/projectnessie/model/Contents.java#L56) when looking it up via `ContentsKey`